pkg/driver/krun/krun_darwin.go

```diff
@@ -0,0 +1,121 @@
+// SPDX-FileCopyrightText: Copyright The Lima Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package krun
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+
+	"github.com/docker/go-units"
+	"github.com/lima-vm/lima/v2/pkg/imgutil/proxyimgutil"
+	"github.com/lima-vm/lima/v2/pkg/iso9660util"
+	"github.com/lima-vm/lima/v2/pkg/limatype"
+	"github.com/lima-vm/lima/v2/pkg/limatype/filenames"
+	"github.com/lima-vm/lima/v2/pkg/limayaml"
+	"github.com/lima-vm/lima/v2/pkg/networks"
+)
+
+const (
+	restfulURI   = "krun-restful.sock" // This is for setting/checking the state of the VM
+	logLevelInfo = "3"
+	KrunEfi      = "krun-efi" // efi variable store
+)
+
+// Cmdline constructs the command line arguments for krunkit based on the instance configuration.
+func Cmdline(inst *limatype.Instance) (*exec.Cmd, error) {
+	var args = []string{
+		"--memory", strconv.Itoa(2048),
+		"--cpus", fmt.Sprintf("%d", *inst.Config.CPUs),
+		"--device", fmt.Sprintf("virtio-serial,logFilePath=%s", filepath.Join(inst.Dir, filenames.SerialLog)),
+		"--krun-log-level", logLevelInfo,
+		"--restful-uri", fmt.Sprintf("unix://%s", restfulSocketPath(inst)),
+		"--bootloader", fmt.Sprintf("efi,variable-store=%s,create", filepath.Join(inst.Dir, KrunEfi)),
+		"--device", fmt.Sprintf("virtio-blk,path=%s,format=raw", filepath.Join(inst.Dir, filenames.DiffDisk)),
+		"--device", fmt.Sprintf("virtio-blk,path=%s", filepath.Join(inst.Dir, filenames.CIDataISO)),
+	}
+
+	// TODO: socket_vmnet and ssh not working
+	networkArgs, err := buildNetworkArgs(inst)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build network arguments: %w", err)
+	}
+	args = append(args, networkArgs...)
+
+	return exec.CommandContext(context.Background(), "krunkit", args...), nil
+}
+
+func buildNetworkArgs(inst *limatype.Instance) ([]string, error) {
+	var args []string
+	nwCfg, err := networks.LoadConfig()
+	if err != nil {
+		return nil, err
+	}
+	socketVMNetOk, err := nwCfg.IsDaemonInstalled(networks.SocketVMNet)
+	if err != nil {
+		return nil, err
+	}
+	if socketVMNetOk {
+		sock, err := networks.Sock("shared")
+		if err != nil {
+			return nil, err
+		}
+		networkArg := fmt.Sprintf("virtio-net,type=unixstream,path=%s,mac=%s,offloading=true",
```

Comment:
offloading=true cannot not work with socket_vment since it does not enable offloading in the vment interface, and we really don't want to enable offloading in vment since it lead to horrible performance.
See https://github.com/containers/krunkit/blob/main/docs/usage.md#offloading-performance-implications

Remove the offloating=true parameter to make the network work. I tested it with socket_vment here:
https://github.com/containers/krunkit/pull/63#issuecomment-3218148429